test/e2e/storage/vsphere_volume_perf.go

```diff
@@ -0,0 +1,179 @@
+	scParamsList = append(scParamsList, scvsanSpec)
+
+	// Create Storage Class with SPBM Policy
+	By("Creating Storage Class : sc-spbm")
+	scSpbmPolicyParameters := make(map[string]string)
+	goldPolicy := os.Getenv("VSPHERE_SPBM_GOLD_POLICY")
+	Expect(goldPolicy).NotTo(BeEmpty())
+	scSpbmPolicyParameters[SpbmStoragePolicy] = goldPolicy
+	scSpbmPolicySpec := getVSphereStorageClassSpec("sc-spbm", scSpbmPolicyParameters)
+	scParamsList = append(scParamsList, scSpbmPolicySpec)
+
+	// Create Storage Class with User Specified Datastore.
+	By("Creating Storage Class : sc-user-specified-datastore")
+	scWithDatastoreParameters := make(map[string]string)
+	datastore := os.Getenv("VSPHERE_DATASTORE")
+	Expect(goldPolicy).NotTo(BeEmpty())
+	scWithDatastoreParameters[Datastore] = datastore
+	scWithDatastoreSpec := getVSphereStorageClassSpec("sc-user-specified-ds", scWithDatastoreParameters)
+	scParamsList = append(scParamsList, scWithDatastoreSpec)
+
+	for _, params := range scParamsList {
+		storageClass, err := client.StorageV1().StorageClasses().Create(params)
+		Expect(err).NotTo(HaveOccurred())
+		scList = append(scList, storageClass)
+	}
+
+	return scList
+}
+
+// invokeVolumeLifeCyclePerformance peforms full volume life cycle management and records latency for each operation
+func invokeVolumeLifeCyclePerformance(f *framework.Framework, client clientset.Interface, namespace string, sc []*storageV1.StorageClass, volumesPerPod int) []int64 {
+	const timeFraction = 1000 // Show results in microseconds
+	var latency []int64
+
+	By(fmt.Sprintf("Creating %v PVCs per pod ", volumesPerPod))
+	var pvclaims []*v1.PersistentVolumeClaim
+	start := time.Now()
+	for i := 0; i < volumesPerPod; i++ {
+		pvclaim, err := framework.CreatePVC(client, namespace, getVSphereClaimSpecWithStorageClassAnnotation(namespace, sc[i%len(sc)]))
+		Expect(err).NotTo(HaveOccurred())
+		pvclaims = append(pvclaims, pvclaim)
+	}
+
+	persistentvolumes, err := framework.WaitForPVClaimBoundPhase(client, pvclaims, framework.ClaimProvisionTimeout)
+	Expect(err).NotTo(HaveOccurred())
+	elapsed := time.Since(start)
+	latency = append(latency, elapsed.Nanoseconds()/timeFraction)
+
+	By("Creating pod to attach PVs to the node and verifying access")
+	start = time.Now()
+	pod, err := framework.CreatePod(client, namespace, pvclaims, false, "")
+	Expect(err).NotTo(HaveOccurred())
+
+	vsp, err := vsphere.GetVSphere()
+	Expect(err).NotTo(HaveOccurred())
+
+	verifyVSphereVolumesAccessible(pod, persistentvolumes, vsp)
+	elapsed = time.Since(start)
+	latency = append(latency, elapsed.Nanoseconds()/timeFraction)
+
+	By("Deleting pod and waiting for volumes to be detached from the node")
+	start = time.Now()
+	err = framework.DeletePodWithWait(f, client, pod)
+	Expect(err).NotTo(HaveOccurred())
+
+	for _, pv := range persistentvolumes {
+		err = waitForVSphereDiskToDetach(vsp, pv.Spec.VsphereVolume.VolumePath, k8stype.NodeName(pod.Spec.NodeName))
```

Comment:
Don't use this implementation. this will take a lot of time to complete. 
Use an implementation for DisksAreAttached()

Reply:
Also waitForVSphereDiskToDetach should not be taken into time consideration. This is a check to just verify. 
DeletePodWithWait would have already done detach of the volume and delete of the pod.